drivers/usb/device/usb_dc_stm32.c

```diff
@@ -437,11 +441,17 @@ static int usb_dc_stm32_init(void)
 	}
 #else /* USB_OTG_FS */
 	/* TODO: make this dynamic (depending usage) */
-	HAL_PCDEx_SetRxFiFo(&usb_dc_stm32_state.pcd, FIFO_EP_WORDS);
+	HAL_PCDEx_SetRxFiFo(&usb_dc_stm32_state.pcd, FIFO_RX_MIN);
 	for (i = 0U; i < USB_NUM_BIDIR_ENDPOINTS; i++) {
-		HAL_PCDEx_SetTxFiFo(&usb_dc_stm32_state.pcd, i,
-				    FIFO_EP_WORDS);
-		k_sem_init(&usb_dc_stm32_state.in_ep_state[i].write_sem, 1, 1);
+		if (i == 0) {
+			/* first endpoint need only 64 byte for EP_TYPE_CTRL */
+			HAL_PCDEx_SetTxFiFo(&usb_dc_stm32_state.pcd, i,	64);
+			k_sem_init(&usb_dc_stm32_state.in_ep_state[i].write_sem, 1, 1);
```

Comment:
`k_sem_init()` can be put after the `if/else`

Reply:
yes